module/os/linux/zfs/arc_os.c

```diff
@@ -126,26 +126,13 @@ arc_available_memory(void)
 	int64_t lowest = INT64_MAX;
 	free_memory_reason_t r = FMR_UNKNOWN;
 	int64_t n;
-	pgcnt_t needfree = btop(arc_need_free);
-	pgcnt_t sysfree = btop(arc_sys_free);
-	pgcnt_t realfree = btop(arc_free_memory());
-
-	if (needfree > 0) {
-		n = PAGESIZE * (-needfree);
-		if (n < lowest) {
-			lowest = n;
-			r = FMR_NEEDFREE;
-		}
+
+	if (arc_need_free > 0) {
+		lowest = -arc_need_free;
+		r = FMR_NEEDFREE;
 	}
 
-	/*
-	 * check that we're out of range of the pageout scanner.  It starts to
-	 * schedule paging if freemem is less than lotsfree and needfree.
-	 * lotsfree is the high-water mark for pageout, and needfree is the
-	 * number of needed free pages.  We add extra pages here to make sure
-	 * the scanner doesn't start up while we're freeing memory.
-	 */
-	n = PAGESIZE * (realfree - sysfree - needfree);
+	n = arc_free_memory() - arc_sys_free;
```

Comment:
It seems like you've removed `arc_need_free` from this calculation, was this intentional?

Reply:
My thinking was that if arc_need_free is nonzero, we'll hit the case above and `lowest` will already be lower than what we could have here.  On second examination, that's only true if `arc_free_memory() - arc_sys_free` is not negative.  So I'll add `arc_need_free` back in here, just in case.